### ISSUE TITLE
Possible break when installing -- call for reproducing

### ISSUE THREAD

**Issue by patrickmj** [omeka/plugin-DublinCoreExtended#17]
With latest Omeka master and latest plugin master, installing gives:

```

Zend_Db_Statement_Mysqli_Exception

    Mysqli statement execute error : Duplicate entry '1-Alternative Title' for key 'name_element_set_id'

Zend_Db_Statement_Mysqli_Exception: Mysqli statement execute error : Duplicate entry '1-Alternative Title' for key 'name_element_set_id' in /var/www/Omeka/application/libraries/Zend/Db/Statement/Mysqli.php:214
Stack trace:
#0 /var/www/Omeka/application/libraries/Zend/Db/Statement.php(303): Zend_Db_Statement_Mysqli->_execute(Array)
#1 /var/www/Omeka/application/libraries/Zend/Db/Adapter/Abstract.php(480): Zend_Db_Statement->execute(Array)
#2 /var/www/Omeka/application/libraries/Omeka/Db.php(79): Zend_Db_Adapter_Abstract->query('\n              ...', Array)
#3 /var/www/Omeka/plugins/DublinCoreExtended/DublinCoreExtendedPlugin.php(58): Omeka_Db->__call('query', Array)
#4 /var/www/Omeka/application/libraries/Omeka/Plugin/Broker.php(145): DublinCoreExtendedPlugin->hookInstall(Array)
#5 /var/www/Omeka/application/libraries/Omeka/Plugin/Installer.php(130): Omeka_Plugin_Broker->callHook('install', Array, Object(Plugin))
#6 /var/www/Omeka/application/controllers/PluginsController.php(97): Omeka_Plugin_Installer->install(Object(Plugin))
#7 /var/www/Omeka/application/libraries/Zend/Controller/Action.php(516): PluginsController->installAction()
#8 /var/www/Omeka/application/libraries/Zend/Controller/Dispatcher/Standard.php(308): Zend_Controller_Action->dispatch('installAction')
#9 /var/www/Omeka/application/libraries/Zend/Controller/Front.php(954): Zend_Controller_Dispatcher_Standard->dispatch(Object(Zend_Controller_Request_Http), Object(Zend_Controller_Response_Http))
#10 /var/www/Omeka/application/libraries/Zend/Application/Bootstrap/Bootstrap.php(105): Zend_Controller_Front->dispatch()
#11 /var/www/Omeka/application/libraries/Zend/Application.php(384): Zend_Application_Bootstrap_Bootstrap->run()
#12 /var/www/Omeka/application/libraries/Omeka/Application.php(77): Zend_Application->run()
#13 /var/www/Omeka/admin/index.php(28): Omeka_Application->run()
#14 {main}
```

It's possible that I have some crazy stuff from importing from piles of sites, so want to ask if others hit this error.

**Comment by mebrett** [comment]
Installed the plugin on my Omeka sites (on master) on both the dev server and my reclaim hosting, got the error `Mysqli statement execute error : Duplicate entry '1-Alternative Title' for key 'name_element_set_id'`

**Comment by zerocrates** [comment]
If I had to guess, I'd guess it was something like a leftover table from a previous install? Anything like that look like it's floating around?

**Comment by patrickmj** [comment]
Not that I see immediately. But on a brand new install it looks like there's no problems.

**Comment by zerocrates** [comment]
I think it must be an "old installation" thing... the only thing the installer does is create the new elements, the only thing the uninstaller does is delete them.

So somehow it would seem they were created and then the plugin was removed, but not uninstalled.

**Comment by patrickmj** [comment]
My bet is the closely related thing of a quirk on API Import. To make sure you can import Item Type Metadata, it has an option to import Element Sets. Thus, importing to a site without DCTerms Extended, from a site with DCTerms Extended, might be creating the DCTerms Extended elements and values, even though DCTerms Extended isn't installed. Thus, going to install it later would lead to that situation of it looking like an old installing, putting the issue in the API Import plugin in the special case of not having all the same plugins installed on both sites. Might be something both for documentation, and for making Import only look at Item Type Metadata, though that might also lead to confusion.

**Comment by patrickmj** [comment]
@mebrett In your installations where that error is happening, does it look like you can use elements from DC Extended anyway (e.g., Temporal Coverage?)

**Comment by zerocrates** [comment]
If the importer pulls in extra elements in existing sets like that, yes, that makes perfect sense as an explanation.

**Comment by mebrett** [comment]
Yes, I do seem to have pulled in extra elements via Omeka API Import.

**Comment by zerocrates** [comment]
The only solution I can think of would be to make the plugin first get all the elements defined currently under Dublin Core in the DB and use that list to decide what, if anything, needs to get added.

That does create the odd scenario though where you could install the plugin, have it add nothing at all, but then if you uninstalled it would delete all the extended DC elements even though it had not created them. The "plugin" model for adding element sets has never been quite right.

**Comment by mebrett** [comment]
Does this mean that an Omeka API import from a site with VRA core would also add VRA core elements without the element set? Or is this a Dublin Core specific quirk?

**Comment by patrickmj** [comment]
It's not Dublin Core specific -- any element set would behave the same way. VRA is special, however, in that only the most minimal, flat, display data is in an element set. All the subtlties are stored separately, and are not currently exposed in the API.

**Comment by jimsafley** [comment]
Fixed in 2ce184fd18c245de50041ac14bd6fff599e59924
